src/OrleansRuntime/Silo/Silo.cs

```diff
@@ -198,14 +198,16 @@ internal Silo(string name, SiloType siloType, ClusterConfiguration config, ILoca
                 LocalDataStoreInstance.LocalDataStore = keyStore;
             }
 
-            services = new DefaultServiceProvider();
+            grainFactory = new GrainFactory();
+            IDictionary<Type,object> singletons = new Dictionary<Type, object> { { typeof(IGrainFactory), grainFactory } };
+            services = new DefaultServiceProvider(singletons);
```

Comment:
>Hmm why not just use an existing container in the meantime?

Doesn't seem worth it to introduce a dependency we don't intend to keep just to save <10 lines of code. :)